x-pack/plugins/index_lifecycle_management/public/application/sections/edit_policy/form/form_errors_context.tsx

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, {
+  createContext,
+  useContext,
+  FunctionComponent,
+  useEffect,
+  useState,
+  useRef,
+} from 'react';
+
+import { useFormData, useFormContext } from '../../../../shared_imports';
+
+import { FormInternal } from '../types';
+
+interface ContextValue {
+  hasErrors: boolean;
+  hot: string[];
+  warm: string[];
+  cold: string[];
+  /**
+   * Errors that are not specific to a phase should go here.
+   */
+  other: string[];
+}
+
+const FormErrorsContext = createContext<ContextValue>(null as any);
+
+const isXPhaseField = (phase: 'hot' | 'warm' | 'cold') => (fieldPath: string): boolean =>
+  fieldPath.startsWith(`phases.${phase}`) || fieldPath.startsWith(`_meta.${phase}`);
+
+const isHotPhaseField = isXPhaseField('hot');
+const isWarmPhaseField = isXPhaseField('warm');
+const isColdPhaseField = isXPhaseField('cold');
+
+const createEmptyContextValue = (): ContextValue => ({
+  hasErrors: false,
+  hot: [],
+  warm: [],
+  cold: [],
+  other: [],
+});
+
+export const FormErrorsProvider: FunctionComponent = ({ children }) => {
+  const [contextValue, setContextValue] = useState<ContextValue>(createEmptyContextValue);
+  const { hasErrors } = contextValue;
+  const isMounted = useRef<boolean>(false);
+
+  const form = useFormContext<FormInternal>();
+
+  // Hook into form updates
+  const [formData] = useFormData<FormInternal>();
+
+  useEffect(() => {
+    isMounted.current = true;
+    return () => {
+      isMounted.current = false;
+    };
+  }, []);
+
+  useEffect(() => {
+    if (!isMounted.current) {
+      return;
+    }
+    // This is a hack, we need to wait for the next tick to let all of the async validation complete
```

Comment:
Is it worth opening up an issue on the form lib side to consider improving this?

Reply:
Hey Alison, that is good point!